internal/pkg/ai-functions/handler/handler_ai_function.go

```diff
@@ -139,6 +92,96 @@ func (h *AIFunction) Apply(ctx context.Context, req *pb.ApplyRequest) (pbValue *
 	return v, nil
 }
 
+func (h *AIFunction) createTestCaseForRequirementIDAndTestID(ctx context.Context, factory functions.FunctionFactory, req *pb.ApplyRequest, openaiURL *url.URL) (any, error) {
+	results := make([]any, 0)
+	var wg sync.WaitGroup
+	var functionParams aitestcase.FunctionParams
+
+	fields := req.GetFunctionParams().GetStructValue().GetFields()
```

Comment:
I think you can use json.marshal&unmarshal to get target structured request.
No magic strings below, it's hard to maintence.

Reply:
Ok